src/include/encoding.h

```diff
@@ -129,6 +125,28 @@ WRITE_INTTYPE_ENCODER(int32_t, le32)
 WRITE_INTTYPE_ENCODER(uint16_t, le16)
 WRITE_INTTYPE_ENCODER(int16_t, le16)
 
+// -----------------------------------
+// float types
+// FIXME: this only performs endian-conversion, but assumes that the
+// floating-point formats are otherwise the same
+
+#define WRITE_FLTTYPE_ENCODER(type, itype, etype)			\
+  inline void encode(type v, ::ceph::bufferlist& bl, uint64_t features=0) { \
+    ceph_assert(sizeof(type) == sizeof(itype));				\
```

Comment:
```suggestion
    static_assert(sizeof(type) == sizeof(itype));				\
```

Reply:
might want to replace the comment of
> but assumes that the
> floating-point formats are otherwise the same

with
```c++
static_assert(std::numeric_limits<v>::is_iec559,
              "float type not using IEEE754");
```

Reply:
Good point.  Updated the assertions.

Reply:
how about moving this static_assert into the macro? less repeating. but not a blocker.

Reply:
I just wanted to have the assert both on little- and big-endian targets.  But if we remove the #ifdef anyway, we can just as well move the assert back into the macro.  Done.